### PR TITLE
:bug: Fix handling pre-releases in mdbook

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:
Fixes an issue where pre-releases would show up in the book.

/assign @ncdc

### CODE REVIEW

File: hack/tools/mdbook/releaselink/releaselink.go [hunk 1]
```diff
@@ -49,7 +50,11 @@ func (l ReleaseLink) Process(input *plugin.Input) error {
 
 		gomodule := tags.Get("gomodule")
 		asset := tags.Get("asset")
-		versionRange := semver.MustParseRange(tags.Get("version"))
+		includePrereleases := tags.Get("prereleases") == "true"
```
Comment (ncdc) [inline hack/tools/mdbook/releaselink/releaselink.go]: Masterminds/semver skips prereleases in its `Contraints` unless the constraint has the prerelease separator + some value. For example, `>= 0.2.0 < 0.3.0` will only look at 0.2.x, whereas `>= 0.2.0 < 0.3.0-0` will check for prereleases too.

Comment (vincepri) [reply hack/tools/mdbook/releaselink/releaselink.go]: I can either switch to the other lib again, or just drop this flag

Comment (ncdc) [reply hack/tools/mdbook/releaselink/releaselink.go]: Drop the flag

**Comment by vincepri** [comment]
/hold cancel

@ncdc went another way, less disruptive change

**Comment by ncdc** [comment]
Preview test 👍 https://5e4c42f2c7419e000712cb88--kubernetes-sigs-cluster-api.netlify.com/user/quick-start.html#install-cluster-api

/lgtm

**Pull request merged**